android/app/src/main/java/com/example/issue_tracker/ui/login/GitHubWebViewFragment.kt

```diff
@@ -0,0 +1,83 @@
+package com.example.issue_tracker.ui.login
```

Comment:
```
package com.example.issue_tracker.ui.login

import android.annotation.SuppressLint
import android.annotation.TargetApi
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.example.issue_tracker.R
import com.example.issue_tracker.common.repeatOnLifecycleExtension
import com.example.issue_tracker.databinding.FragmentGitHubWebViewBinding
import com.example.issue_tracker.model.GitHubOAuthRequest
import com.example.issue_tracker.ui.HomeActivity
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.flow.collect

@AndroidEntryPoint
class GitHubWebViewFragment : Fragment() {

    lateinit var binding: FragmentGitHubWebViewBinding
    private lateinit var navigator: NavController
    private val viewModel: LoginViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        binding =
            DataBindingUtil.inflate(inflater, R.layout.fragment_git_hub_web_view, container, false)
        binding.lifecycleOwner = viewLifecycleOwner
        return binding.root
    }

    @SuppressLint("SetJavaScriptEnabled")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        navigator = Navigation.findNavController(view)
        binding.githubWebView.run {
            webViewClient = CustomWebViewClient()
            loadUrl("https://github.com/login/oauth/authorize?client_id=" + getString(R.string.git_hub_id))
        }

        setupObservers()
    }

    private fun setupObservers() {
        viewLifecycleOwner.repeatOnLifecycleExtension {
            viewModel.accessToken.collect { accessToken ->
                if (!accessToken.isNullOrEmpty()) {
                    val intent = Intent(requireContext(), HomeActivity::class.java)
                    startActivity(intent)
                }
            }
        }
    }

    inner class CustomWebViewClient : WebViewClient() {

        private fun checkUrl(request: WebResourceRequest?) {
            val url = request?.url ?: return
            val code = url.getQueryParameter(GUTHUB_OAUTH_CODE_PARAM_KEY)

            if (url.scheme == GUTHUB_OAUTH_REDIRECTION_SCHEME &&
                url.host == GUTHUB_OAUTH_REDIRECTION_HOST &&
                code != null
            ) {
                viewModel.requestGitHubLogin(GitHubOAuthRequest(code))
            }
        }

        @TargetApi(Build.VERSION_CODES.N)
        override fun shouldOverrideUrlLoading(
            view: WebView?,
            request: WebResourceRequest?,
        ): Boolean {
            checkUrl(request)
            return true
        }
    }

    companion object {
        private const val GUTHUB_OAUTH_REDIRECTION_SCHEME = "http"
        private const val GUTHUB_OAUTH_REDIRECTION_HOST = "13.209.73.68"
        private const val GUTHUB_OAUTH_CODE_PARAM_KEY = "code"
    }
}
```

url 프로토콜 형식에 맞게 검증하도록 개선하는게 좋아보입니다.

string 비교로 하면 코드 의도를 알기도 힘들고 url이 변경 됐을때 대응도 쉽지 않을것 같아요.

shouldOverrideUrlLoading 함수 반환 값을 return true로 하면 다음 page를 load 하지 않아요.

redirection 된 페이지 갈 필요 없이 앱에서 처리할거니 위처럼 하면 될겁니다.
(질문에 올리신 문제도 이걸로 해결될거로 보입니다.)

```
Returns: true to cancel the current load, otherwise return false.
```